azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java

```diff
@@ -440,24 +447,16 @@ protected void listHTTPTriggerUrls(IFunctionApp target) {
     }
 
     private List<FunctionEntity> listFunctions(final IFunctionApp functionApp) {
-        for (int i = 0; i < LIST_TRIGGERS_MAX_RETRY; i++) {
-            try {
-                AzureMessager.getMessager().info(String.format(SYNCING_TRIGGERS_AND_FETCH_FUNCTION_INFORMATION, i + 1, LIST_TRIGGERS_MAX_RETRY));
-                functionApp.syncTriggers();
-                final List<FunctionEntity> triggers = functionApp.listFunctions();
-                if (CollectionUtils.isNotEmpty(triggers)) {
-                    return triggers;
-                }
-            } catch (RuntimeException e) {
-                // swallow service exception while list triggers
-            }
-            try {
-                Thread.sleep(LIST_TRIGGERS_RETRY_PERIOD_IN_SECONDS * 1000);
-            } catch (InterruptedException e) {
-                // swallow interrupted exception
-            }
-        }
-        throw new AzureToolkitRuntimeException(NO_TRIGGERS_FOUNDED);
+        final AtomicInteger count = new AtomicInteger(0);
+        return Mono.fromCallable(() -> {
+            final AzureString message = count.getAndAdd(1) == 0 ?
+                    AzureString.fromString(SYNCING_TRIGGERS) : AzureString.format(SYNCING_TRIGGERS_WITH_RETRY, count.get(), LIST_TRIGGERS_MAX_RETRY);
+            AzureMessager.getMessager().info(message);
+            return Optional.ofNullable(functionApp.listFunctions(true))
+                    .filter(CollectionUtils::isNotEmpty)
+                    .orElseThrow(() -> new AzureToolkitRuntimeException(NO_TRIGGERS_FOUNDED));
+        }).subscribeOn(Schedulers.boundedElastic())
+                .retryWhen(Retry.backoff(LIST_TRIGGERS_MAX_RETRY - 1, Duration.ofSeconds(LIST_TRIGGERS_RETRY_PERIOD_IN_SECONDS))).block();
```

Comment:
will `block()` throw exception `NO_TRIGGERS_FOUNDED` if the last try still fails?

Reply:
Will throw `reactor.core.Exceptions$RetryExhaustedException` while no triggers found was the cause. And this exception will be caught by called and show warning for users